frontend/src/pages/analytics/segments/components/group/ConditionTimeRange.tsx

```diff
@@ -0,0 +1,135 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import {
+  DateRangePicker,
+  DateRangePickerProps,
+} from '@cloudscape-design/components';
+import { IEventSegmentationItem } from 'components/eventselect/AnalyticsType';
+import { AnalyticsSegmentActionType } from 'components/eventselect/reducer/analyticsSegmentGroupReducer';
+import { useSegmentContext } from 'context/SegmentContext';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { defaultStr } from 'ts/utils';
+
+interface ConditionTimeRangeProps {
+  segmentData: IEventSegmentationItem;
+  groupIndex: number;
+}
+
+const ConditionTimeRange: React.FC<ConditionTimeRangeProps> = (
+  props: ConditionTimeRangeProps
+) => {
+  const { t } = useTranslation();
+  const { segmentData, groupIndex } = props;
+  const { segmentDataDispatch } = useSegmentContext();
+
+  const isValidRange = (
+    range: DateRangePickerProps.Value | null
+  ): DateRangePickerProps.ValidationResult => {
+    if (range?.type === 'absolute') {
+      const [startDateWithoutTime] = range.startDate.split('T');
+      const [endDateWithoutTime] = range.endDate.split('T');
+      if (!startDateWithoutTime || !endDateWithoutTime) {
+        return {
+          valid: false,
+          errorMessage: t('analytics:valid.dateRangeIncomplete'),
+        };
+      }
+      if (
+        new Date(range.startDate).getTime() -
+          new Date(range.endDate).getTime() >
+        0
+      ) {
+        return {
+          valid: false,
+          errorMessage: t('analytics:valid.dateRangeInvalid'),
+        };
+      }
+    }
+    return { valid: true };
+  };
+  return (
+    <div>
+      <DateRangePicker
+        onChange={({ detail }) => {
+          console.log(detail.value);
+          segmentDataDispatch({
+            type: AnalyticsSegmentActionType.UpdateFilterGroupTimeRange,
+            index: groupIndex,
+            timeRange: detail,
+          });
+        }}
+        value={segmentData.groupDateRange?.value ?? null}
+        relativeOptions={[
+          {
+            key: 'previous-5-minutes',
+            amount: 5,
+            unit: 'minute',
+            type: 'relative',
+          },
+          {
+            key: 'previous-30-minutes',
+            amount: 30,
+            unit: 'minute',
+            type: 'relative',
+          },
+          {
+            key: 'previous-1-hour',
+            amount: 1,
+            unit: 'hour',
+            type: 'relative',
+          },
+          {
+            key: 'previous-6-hours',
+            amount: 6,
+            unit: 'hour',
+            type: 'relative',
+          },
+        ]}
```

Comment:
Are they all the options?

Reply:
Updated same with analytics discovery relative time options.